Basic-Scripts/binary_search.py

```diff
@@ -0,0 +1,37 @@
+def bin_search(n,s):
+    low = 0                             # initialize left index
+    high = len(n) - 1                   # initialize right index
+
+    while low <= high:
+        index = (low + high)//2         # find the mid-point of the list
+
+        if n[index] < s:                # element is in upper half so ignore lower half
+            low = index + 1
+        elif n[index] > s:              # element is in lower half so ignore upper half
+            high = index - 1
+        else:
+            return index
+
+    return -1                           # if we reach this part of the code, it means the element does not exist
+
+print("Enter the number of numbers you want to search:")
+num = int(input())
+
+n = []     # define a new list to hold the numbers to be searched
```

Comment:
Use meaningful variable names instead of alphabets.
Include __name__ == "__main__".
Include docstrings for ur functions